cli/src/main.rs

```diff
@@ -580,53 +580,118 @@ fn run<I: IntoIterator<Item = T>, T: Into<OsString> + Clone>(args: I) -> Result<
+                        .help("Path of public key file to include with node"),
+                )
+                .arg(
+                    Arg::with_name("metadata")
+                        .long("metadata")
+                        .takes_value(true)
+                        .multiple(true)
+                        .help("Metadata to include with node (<key>=<value>)"),
+                )
+                .arg(
+                    Arg::with_name("private_key_file")
+                        .value_name("private-key-file")
+                        .short("k")
+                        .long("key")
+                        .takes_value(true)
+                        .help("Name or path of private key"),
+                ),
+        );
+
+    #[cfg(feature = "registry-add-node")]
+    let registry_command = registry_command.subcommand(
+        SubCommand::with_name("add")
+            .about("Add a node to the local registry")
+            .arg(
+                Arg::with_name("display_name")
+                    .long("display")
+                    .takes_value(true)
+                    .help("Human-readable name for the new node"),
+            )
+            .arg(
+                Arg::with_name("dry_run")
+                    .long("dry-run")
+                    .help("Show the expected changes without submitting the node"),
+            )
+            .arg(
+                Arg::with_name("endpoint")
+                    .long("endpoint")
+                    .takes_value(true)
+                    .multiple(true)
+                    .required_unless("from_remote")
+                    .help("Network endpoint for the new node"),
+            )
+            .arg(
+                Arg::with_name("from_remote")
+                    .long("from-remote")
+                    .help("Copies an existing node definition from the remote registries"),
+            )
+            .arg(
+                Arg::with_name("identity")
+                    .required(true)
+                    .help("Identity of the new node. Must be unique in the local registry"),
+            )
+            .arg(
+                Arg::with_name("key_files")
+                    .long("key-file")
+                    .takes_value(true)
+                    .multiple(true)
+                    .required_unless("from_remote")
+                    .help("Path of public key file to include with node"),
+            )
+            .arg(
+                Arg::with_name("metadata")
+                    .long("metadata")
+                    .takes_value(true)
+                    .multiple(true)
+                    .help("Metadata to include with node (<key>=<value>)"),
+            )
+            .arg(
+                Arg::with_name("private_key_file")
+                    .value_name("private-key-file")
+                    .short("k")
+                    .long("key")
+                    .takes_value(true)
+                    .help("Name or path of private key"),
```

Comment:
Might want to make this help text a bit more explicit so it's clear that this doesn't get attached to the node in any way.

Reply:
Updated